src/BlogDetail.tsx

```
@@ -0,0 +1,38 @@
+
+  if (loading) return <div className="min-h-screen flex items-center justify-center text-gray-500">Loading...</div>;
+  if (!blog) return <div className="min-h-screen flex items-center justify-center text-gray-400">Blog post not found.</div>;
+
+  return (
+    <div className="max-w-2xl mx-auto py-10 px-4 min-h-screen bg-gradient-to-br from-gray-50 to-gray-100">
+      <Link to="/blog" className="text-blue-600 hover:underline text-sm mb-4 inline-block">← Back to Blog</Link>
+      {blog.imageUrl && <img src={blog.imageUrl} alt={blog.title} className="w-full h-60 object-cover rounded mb-6" />}
+      <div className="text-xs text-gray-400 mb-1">{blog.createdAt && blog.createdAt.toDate().toLocaleDateString()}</div>
+      <h1 className="text-3xl font-bold mb-4 text-gray-900">{blog.title}</h1>
+      <div className="text-gray-600 mb-4 text-lg">{blog.summary}</div>
+      <div className="prose prose-lg max-w-none text-gray-800" dangerouslySetInnerHTML={{ __html: blog.content.replace(/\n/g, '<br/>') }} />
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)